### ISSUE TITLE
Ampersand is decoded while parsing and not encoded back when bulding 

### ISSUE THREAD

**Issue by GeorgeTailor** [NaturalIntelligence/fast-xml-parser#418]
* [x] Are you running the latest version?
* [x] Have you included sample input, output, error, and expected output? 
* [x] Have you checked if you are using correct configuration?
* [x] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

### Input
```xml
<score-partwise version="2.0">
 <identification>
  <creator type="transcriber">Guitars: Luke Slater, Bass &amp; Drums: Sean Loseby &amp; TROLL for the basic lyrics and a few drum corrections.</creator>
</identification>
</score-partwise>
```

### Code
```js
const parser = new XMLParser({
	ignoreAttributes : false,
	preserveOrder: true
});
const parsedData = parser.parse(data);

...

const builder = new XMLBuilder({
	ignoreAttributes: false,
	preserveOrder: true
});
const xml = builder.build(parsedData);
```

### Output

```xml
<score-partwise version="2.0">
<identification>
<creator type="transcriber">Guitars: Luke Slater, Bass & Drums: Sean Loseby & TROLL for the basic lyrics and a few drum corrections.</creator>
</identification>
</score-partwise>
```

When I try to validate the resulting XML with 
```js
const result = XMLValidator.validate(xml, {
	allowBooleanAttributes: true
});
```
I also get the following error 
```json
{
  err: {
    code: 'InvalidChar',
    msg: "char '&' is not expected.",
    line: 1,
    col: 359
  }
}
```

### expected data
`&` should stay encoded as there were in the original file. & is an invalid character in the xml and the document becomes unusable.

**Would you like to work on this issue?**

- [ ] Yes
- [x] No

**Comment by amitguptagwl** [comment]
`&` us generally encoded so that it can't be mixed with other char sequence to form some entity. However, this is not the case with FXP. Hence, it is not implemented. But since it is a second issue, I'll push it in queue to handle this.

**Comment by GeorgeTailor** [comment]
Similar case is for `"`, original XML has plain `"` sign, but when parsed -> built it becomes `&quot;`. Should I create another ticket for that?

**Comment by amitguptagwl** [comment]
well! `'`, `"`, `>`, `<` are already handled by the builder. I hope you're using `processEntities: true`. Please check builder docs once.

**Comment by GeorgeTailor** [comment]
but the docs mention that `processEntities` is set to true by default https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/v4/3.XMLBuilder.md#processentities

**Comment by amitguptagwl** [comment]
Yes, it is `true` by default. Let me check why it is breaking for you.

**Comment by patryk-smc** [comment]
I believe I have a similar issue.

```
{
  "people": {
    "person": [
      {
        "name": "John"
      },
      {
        "name": "Molly"
      }
    ]
  }
}
```

Gets parsed to:

```
<?xml version="1.0"?>
<people>
  <person>
    <name>John</name>
  </person>
  <person>
    <name>Molly</name>
  </person>
</people>
```

Which is valid XML.

But once I add an ampersand:

```
{
  "people": {
    "person": [
      {
        "name": "John & John"
      },
      {
        "name": "Molly"
      }
    ]
  }
}
```

I get this:

```
<?xml version="1.0"?>
<people>
  <person>
    <name>John & John</name>
  </person>
  <person>
    <name>Molly</name>
  </person>
</people>
```

Opening such XML file in Chrome or Safari results in error:

<img width="556" alt="CleanShot 2022-01-17 at 17 39 43@2x" src="https://user-images.githubusercontent.com/37963339/149808279-d4e99164-6368-4226-9713-ad3fe55457da.png">

**Comment by josephdpurcell** [comment]
By setting `processEntities: false` I was able to avoid `&amp;` being converted to `&`.

**Comment by amitguptagwl** [comment]
This issue was fixed. @josephdpurcell are you facing any issue now? if yes can you please confirm if you're using latest version of the package?

**Comment by josephdpurcell** [comment]
Great question! I am using `fast-xml-parser` version `4.0.1`. I see the current version is `4.0.7`.

I see in the release notes for `4.0.2`:

```
* builder supports decoding &
```

My conclusion would be that I am experiencing the issue because I do not have the fixes from `4.0.2`. If I upgrade and continue seeing this issue I'll be sure to report back!
